src/ExcelProvider.Runtime/ExcelProvider.Runtime.fs

```diff
@@ -329,16 +338,27 @@ type Row(documentId, sheetname, rowIndex, getCellValue: int -> int -> obj, colum
             |> Seq.map (fun kvp -> kvp.Key)
             |> Seq.tryFind (fun header -> String.Equals(header, columnName, StringComparison.OrdinalIgnoreCase))
             |> function
-                | Some header -> sprintf "ExcelProvider: Column \"%s\" was not found. Did you mean \"%s\"?" columnName header
+                | Some header ->
+                    sprintf "ExcelProvider: Column \"%s\" was not found. Did you mean \"%s\"?" columnName header
                 | None -> sprintf "ExcelProvider: Column \"%s\" was not found." columnName
             |> failwith
 
     member this.TryGetValue<'a> (columnIndex: int) columnName =
         let value = this.GetValue columnIndex
 
         try
-            value :?> 'a
-        with :? InvalidCastException ->
+            match value with
+            | null -> value :?> 'a
+            | _ when typeof<'a> = typeof<string> -> (box (string value)) :?> 'a
+            //| :? string as valueStr when typeof<'a> = typeof<double> -> (box (Double.Parse valueStr)) :?> 'a
```

Comment:
This line didn't seem to change the default behavior, so I left it commented out. It could be added if we wanted to be more explicit, or find a case this fails on. Otherwise, fine to remove it.